glean-core/src/metrics/time_unit.rs

```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Serialize, Deserialize};
+use std::convert::TryFrom;
+
+use crate::error::{Error, ErrorKind};
+
+/// Enumeration of different resolutions supported by the
+/// time related metric types (e.g. DatetimeMetric).
+//#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum TimeUnit {
+    Nanosecond,
+    Microsecond,
+    Millisecond,
+    Second,
+    Minute,
+    Hour,
+    Day
+}
+
+impl TimeUnit {
+    /// TODO
+    pub fn format_pattern(&self) -> &'static str {
+        use TimeUnit::*;
+        match self {
+            Nanosecond => "%Y-%m-%dT%H:%M:%S.%.2f%z",
```

Comment:
```suggestion
            Nanosecond => "%Y-%m-%dT%H:%M:%S%.3f%z",
```

3 digits, includes the dot.
See https://docs.rs/chrono/0.4.6/chrono/format/strftime/index.html

Reply:
Thank you! That was it :) I had to slightly change the enums though, to support different resolution.